File: stack/stack.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "stack.h"
#include "../linked_list/linked_list.h"

Stack* CreateStack(int capacity) {
  Stack* stack = (Stack*)malloc(sizeof(Stack));
  stack->top = NULL;
  stack->capacity = capacity;
  stack->linked_list = (void*)malloc(stack->capacity * sizeof(void*));
  return stack;
}

void PushToStack(Stack* stack, void* data) {
  Push(stack->linked_list, data);
  stack->capacity = stack->capacity + 1;
}

void* PopFromStack(Stack* stack) {
  void* data = Pop(stack->linked_list);
  return data;
}

void PrintStack(Stack* stack) {
  PrintLinkedList(stack->linked_list, PrintInt);
}

void DestroyStack(Stack* stack) {
  Destroy(stack->linked_list);
  free(stack);
}

File: stack/stack.py
import sys
sys.path.append("../linked_list")
from linked_list import * 

class Stack:

    def __init__(self, capacity):
        self.capacity = capacity
        self.linked_list = LinkedList()

    def pop(self):
        return self.linked_list.pop().data
    
    def push(self, data):
        self.linked_list.push(data)

    def print_stack(self):
        self.linked_list.print_linked_list()
